micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -46,19 +46,25 @@ impl NVSStorage {
         })
     }
 
-    fn get_string(&self, key: &'static str) -> Result<String, NVSStorageError> {
+    fn get_string(&self, key: &str) -> Result<String, NVSStorageError> {
         let nvs = self.nvs.borrow_mut();
         let len = nvs
             .str_len(key)?
-            .ok_or(NVSStorageError::NVSKeyAbsent(key))?;
+            .ok_or(NVSStorageError::NVSKeyAbsent(key.to_string()))?;
         let mut buf = vec![0_u8; len];
         Ok(nvs
             .get_str(key, buf.as_mut_slice())?
-            .ok_or(NVSStorageError::NVSKeyAbsent(key))?
+            .ok_or(NVSStorageError::NVSKeyAbsent(key.to_string()))?
             .to_owned())
     }
 
     fn set_string(&self, key: &str, string: &str) -> Result<(), NVSStorageError> {
+        if self.has_string(key).unwrap_or_default()
+            && self.get_string(key).unwrap_or("".to_string()).as_str() == string
```

Comment:
can you not do `self.get_string(key).unwrap_or_default().as_str()`, like what you do with `has_string`?